task.go

```diff
@@ -226,9 +259,22 @@ func (task *Task) IsTerminated() bool {
 	return IsTerminal(st)
 }
 
-// IsWaiting returns whether a task is waiting
-func (task *Task) IsWaiting() bool {
-	return task.CurrentStatus() == TaskQueued
+// IsActive returns whether the task is still active.
+func (task *Task) IsActive() bool {
+	st := task.CurrentStatus()
+	if st == "" {
```

Comment:
This case is already covered by IsActive() because they are all checking for an explicit inclusion in a set so lets drop this extra check

Reply:
yep good point, same for queued. Thanks!